File: CS/GrowthWare.WebApplication/UserControls/GroupControl.ascx.cs
using System;
using System.Collections;
using System.Web.UI;

namespace GrowthWare.WebApplication.UserControls
{
    public partial class GroupControl : UserControl
    {
        #region Pulbic Properties
        public ArrayList AllGroups;
        public Array SelectedViewGroups;
        public Array SelectedAddGroups;
        public Array SelectedEditGroups;
        public Array SelectedDeleteGroups;
        public string ViewGroups
        {
            get { return ctlViewGroups.SelectedState; }
        }
        public bool ViewGroupsChanged
        {
            get { return ctlViewGroups.Changed; }
        }
        public string AddGroups
        {
            get { return ctlAddGroups.SelectedState; }
        }
        public bool AddGroupsChanged
        {
            get { return ctlAddGroups.Changed; }
        }
        public string EditGroups
        {
            get { return ctlEditGroups.SelectedState; }
        }
        public bool EditGroupsChanged
        {
            get { return ctlEditGroups.Changed; }
        }
        public string DeleteGroups
        {
            get { return ctlDeleteGroups.SelectedState; }
        }
        public bool DeleteGroupsChanged
        {
            get { return ctlDeleteGroups.Changed; }
        }
        #endregion

        #region Page Event Handlers
        protected void Page_Load(object sender, System.EventArgs e)
        {
            ctlViewGroups.DataSource = AllGroups;
            ctlAddGroups.DataSource = AllGroups;
            ctlEditGroups.DataSource = AllGroups;
            ctlDeleteGroups.DataSource = AllGroups;
            if ((SelectedViewGroups != null)) ctlViewGroups.SelectedItems = (string[])SelectedViewGroups;
            if ((SelectedAddGroups != null)) ctlAddGroups.SelectedItems = (string[])SelectedAddGroups;
            if ((SelectedEditGroups != null)) ctlEditGroups.SelectedItems = (string[])SelectedEditGroups;
            if ((SelectedDeleteGroups != null)) ctlDeleteGroups.SelectedItems = (string[])SelectedDeleteGroups;
            DataBind();
        }
        #endregion
    }
}

File: VB/GrowthWare.WebApplication/UserControls/GroupControl.ascx.vb
﻿Public Class GroupControl
    Inherits System.Web.UI.UserControl

    Public AllGroups As ArrayList
    Public SelectedViewGroups As Array
    Public SelectedAddGroups As Array
    Public SelectedEditGroups As Array
    Public SelectedDeleteGroups As Array

    Public ReadOnly Property ViewGroups() As String
        Get
            Return ctlViewGroups.SelectedState
        End Get
    End Property

    Public ReadOnly Property ViewGroupsChanged() As Boolean
        Get
            Return ctlViewGroups.Changed
        End Get
    End Property

    Public ReadOnly Property AddGroups() As String
        Get
            Return ctlAddGroups.SelectedState
        End Get
    End Property

    Public ReadOnly Property AddGroupsChanged() As Boolean
        Get
            Return ctlAddGroups.Changed
        End Get
    End Property

    Public ReadOnly Property EditGroups() As String
        Get
            Return ctlEditGroups.SelectedState
        End Get
    End Property

    Public ReadOnly Property EditGroupsChanged() As Boolean
        Get
            Return ctlEditGroups.Changed
        End Get
    End Property

    Public ReadOnly Property DeleteGroups() As String
        Get
            Return ctlDeleteGroups.SelectedState
        End Get
    End Property

    Public ReadOnly Property DeleteGroupsChanged() As Boolean
        Get
            Return ctlDeleteGroups.Changed
        End Get
    End Property

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        ctlViewGroups.DataSource = AllGroups
        ctlAddGroups.DataSource = AllGroups
        ctlEditGroups.DataSource = AllGroups
        ctlDeleteGroups.DataSource = AllGroups
        If Not SelectedViewGroups Is Nothing Then ctlViewGroups.SelectedItems = SelectedViewGroups
        If Not SelectedAddGroups Is Nothing Then ctlAddGroups.SelectedItems = SelectedAddGroups
        If Not SelectedEditGroups Is Nothing Then ctlEditGroups.SelectedItems = SelectedEditGroups
        If Not SelectedDeleteGroups Is Nothing Then ctlDeleteGroups.SelectedItems = SelectedDeleteGroups
        DataBind()
    End Sub

End Class